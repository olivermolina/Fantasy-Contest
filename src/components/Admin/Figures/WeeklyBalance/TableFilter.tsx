import React from 'react';
import { Button, Checkbox, FormControlLabel, TextField } from '@mui/material';
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { MobileDatePicker } from '@mui/x-date-pickers/MobileDatePicker';
import DebouncedInput from '~/components/ContestDetail/Entries/DebouncedInput';
import dayjs, { Dayjs } from 'dayjs';
import weekday from 'dayjs/plugin/weekday';
import updateLocale from 'dayjs/plugin/updateLocale';

dayjs.extend(updateLocale);
dayjs.updateLocale('en', {
  weekStart: 1,
});

dayjs.extend(weekday);

interface TableFilterProps {
  setDate: (value: ((prevState: Date) => Date) | Date) => void;
  globalFilter: string;
  setGlobalFilter: (value: ((prevState: string) => string) | string) => void;
  viewInactive: boolean;
  setViewInactive: (value: ((prevState: boolean) => boolean) | boolean) => void;
}

export default function TableFilter(props: TableFilterProps) {
  const { setGlobalFilter, globalFilter, setDate } = props;
  const [value, setValue] = React.useState<Dayjs | null>();
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    props.setViewInactive(event.target.checked);
  };

  const weeksAgoOnClick = (week: number) => {
    if (week === 0) {
      setDate(new Date());
      return;
    }
    const date = dayjs(new Date()).weekday(-(7 * week));
    setDate(date.toDate());
    setValue(date);
  };
  return (
    <div className={'flex flex-col gap-2 py-2'}>
      <div
        className={'flex flex-col md:flex-row gap-2 w-full lg:w-3/4 xl:w-1/2'}
      >
        <LocalizationProvider dateAdapter={AdapterDayjs}>
          <MobileDatePicker
            label="Day Week Date"
            renderInput={(params) => <TextField {...params} />}
            value={value}
            onChange={(newValue) => {
              setValue(newValue);
              if (newValue) {
                setDate(newValue.toDate());
              }
            }}
          />
        </LocalizationProvider>
        <DebouncedInput
          value={globalFilter ?? ''}
          onChange={(value) => {
            setGlobalFilter(String(value));
          }}
          placeholder={'Search'}
          debounce={200}
        />
        <FormControlLabel
          label={'View inactive only'}
          control={
            <Checkbox
              checked={props.viewInactive}
              onChange={handleChange}
              inputProps={{ 'aria-label': 'controlled' }}
            />
          }
        />
      </div>
      <div className={'flex flex-row-reverse gap-2 justify-end'}>
        {[...Array(9)].map((row, index) => (
          <Button
            key={index}
            variant="contained"
            onClick={() => weeksAgoOnClick(index)}
            sx={{ minnWidth: 130 }}
          >
            {index > 1 ? `${index} weeks ago` : null}
            {index === 1 ? 'Last week' : null}
            {index === 0 ? 'This week' : null}
          </Button>
        ))}
      </div>
    </div>
  );
}
