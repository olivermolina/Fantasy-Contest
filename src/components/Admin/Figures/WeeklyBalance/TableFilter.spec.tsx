import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import TableFilter from './TableFilter';

describe('TableFilter', () => {
  it('should render correctly', () => {
    const props = {
      setGlobalFilter: jest.fn(),
      setDate: jest.fn(),
      globalFilter: '',
      viewInactive: false,
      setViewInactive: jest.fn(),
    };

    const { container } = render(<TableFilter {...props} />);
    expect(container).toMatchSnapshot();
  });
});
