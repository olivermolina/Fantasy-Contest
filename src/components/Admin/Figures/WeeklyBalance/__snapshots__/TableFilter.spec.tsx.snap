// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableFilter should render correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2 py-2"
  >
    <div
      class="flex flex-col md:flex-row gap-2 w-full lg:w-3/4 xl:w-1/2"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for=":r0:"
          id=":r0:-label"
        >
          Day Week Date
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl Mui-readOnly css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            aria-label="Choose date"
            aria-readonly="true"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-readOnly css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            readonly=""
            type="text"
            value="01/01/2021"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Day Week Date
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="relative grow"
      >
        <div
          class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
        >
          <svg
            aria-hidden="true"
            class="w-5 h-5 text-gray-500 dark:text-gray-400"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            />
          </svg>
        </div>
        <input
          class="block p-4 pl-10 w-full text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-2 focus:ring-blue-500 focus:border-blue-500 focus:outline-none"
          placeholder="Search"
          required=""
          type="search"
          value=""
        />
      </div>
      <label
        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            aria-label="controlled"
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          View inactive only
        </span>
      </label>
    </div>
    <div
      class="flex flex-row-reverse gap-2 justify-end"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        This week
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Last week
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        2 weeks ago
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        3 weeks ago
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        4 weeks ago
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        5 weeks ago
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        6 weeks ago
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        7 weeks ago
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1rxu0pa-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        8 weeks ago
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
